src/soar_sdk/views/template_filters.py

```diff
@@ -0,0 +1,223 @@
+import hashlib
+import json
+import re
+import uuid
+import humanize
+import bleach  # type: ignore[import-untyped]
+from datetime import datetime, timedelta
+from typing import Optional, Union
+from collections.abc import Iterator
+from jinja2 import Environment
+
+
+def widget_uuid(length: int = 8) -> str:
+    """Generate a unique widget ID."""
+    unique_id = f"widget_{uuid.uuid4()}"
+    return unique_id.replace("-", "")
+
+
+def datetime_minutes(dt: timedelta) -> int:
+    """Convert datetime to minutes."""
+    return (dt.days * 24 * 60) + (dt.seconds // 60)
+
+
+def human_datetime(value: datetime, relative: bool = True) -> str:
+    """Format datetime in human readable format."""
+    if humanize and relative:
+        return humanize.naturaltime(value)
+    else:
+        return "{dt:%b} {dt:%-d}, {dt:%Y} {dt:%-I}:{dt:%M} {ampm}".format(
+            dt=value, ampm=value.strftime("%p").lower()
+        )
+
+
+def human_timedelta(td: timedelta) -> str:
+    """Format timedelta in human readable format."""
+    if humanize:
+        return humanize.naturaldelta(td)
+    return str(td)
+
+
+def sorteditems(dictionary: dict) -> list:
+    """Sort dictionary items by key."""
+    result = []
+    for key in sorted(dictionary):
+        result.append((key, dictionary[key]))
+    return result
+
+
+def batch(iterable: list, count: int) -> Iterator[list]:
+    """Batch items into groups of specified count."""
+    result = []
+    for item in iterable:
+        result.append(item)
+        if len(result) == count:
+            yield result
+            result = []
+    if result:
+        yield result
+
+
+def dict_batch(d: dict, count: int) -> Iterator[list]:
+    """Batch dictionary items into groups."""
+    result = []
+    for key, value in d.items():
+        result.append((key, value))
+        if len(result) == count:
+            yield result
+            result = []
+    if result:
+        yield result
+
+
+def remove_empty(dictionary: dict) -> dict:
+    """Remove empty values from dictionary."""
+    return {k: v for k, v in dictionary.items() if v}
+
+
+def by_key(dictionary: dict, key: str) -> str:
+    """Get dictionary value by key."""
+    return dictionary.get(key, "")
+
+
+def by_nested_key(dictionary: dict, key: str) -> Optional[str]:
+    """Get nested dictionary value by space-separated key."""
+    split_key = key.split()
+    src = dictionary.get(split_key[0], None)
+    if src:
+        return src.get(split_key[1], "")
+    else:
+        return None
+
+
+def is_list(item: object) -> bool:
+    """Check if item is a list."""
+    return isinstance(item, list)
+
+
+def typeof(item: object) -> type:
+    """Get type of item."""
+    return type(item)
+
+
+def safe_intcomma(value: Union[str, int]) -> str:
+    """Format integer with commas, safely handling non-integers."""
+    try:
+        int(value)
+        if humanize:
+            return humanize.intcomma(value)
+        else:
+            return f"{int(value):,}"
```

Comment:
Since this is part of the f-string standard as of Python 3.6, we can just use it without humanize:
```suggestion
        return f"{int(value):,}"
```